extra_geom/base.py

```diff
@@ -794,6 +794,77 @@ def data_coords_to_positions(self, module_no, slow_scan, fast_scan):
             + (np.expand_dims(tile_ss, -1) * coords_ss_vec) \
             + (np.expand_dims(tile_fs, -1) * coords_fs_vec)
 
+    def positions_to_data_coords(self, positions,  origin=None, unit=None):
+        """Convert assembled positions to data array coordinates
+
+        Data array coordinates are how you might refer to a pixel in an array of
+        detector data: module number, and indices in the slow-scan and fast-scan
+        directions. But coordinates in the two pixel dimensions aren't necessarily
+        integers, e.g. if they refer to the centre of a peak.
+
+        Positions is a 2D array (n, 2) describing (x, y) positions of points within an
+        assembled detector image. The may hold floats for sub-pixel positions.
+
+        If origin is None (default) the position indices are relative to the centre of
+        the geometry. `origin=(0, 0)` would set the origin to the first index of the
+        assembled array.
+        If unit is None (default) the positions given are in pixel, else given positions
+        are in meter relative to the centre, e.g. `unit=1e-3` interprets positions in mm.
```

Comment:
For new methods like this, where we can be consistent across all detectors, how would you feel about saying the input is always in metres, and if you've got e.g. data in pixel units, you apply the unit like `data * geom.pixel_size`?